scipy/optimize/tests/test_minpack.py

```diff
@@ -517,6 +517,131 @@ def test_maxfev_and_bounds(self):
         assert_allclose(popt1, 2, atol=1e-14)
         assert_allclose(popt2, 2, atol=1e-14)
 
+    def test_curvefit_simplecovariance_nojacobian(self):
```

Comment:
The new tests are aimed to check matrix `sigma` arguments, which should work for any combinations of other arguments. The most important among them is `jac`, so we check for both variants of `jac`.

It's just infeasible to follow the approach where each variation is done in a separate function. The size of unit tests would increase multiple times if we would follow that.